src/Raven.Server/Documents/Handlers/Processors/Replication/AbstractReplicationHandlerProcessorForGetOutgoingFailureStats.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Replication;
+using Raven.Client.Http;
+using Raven.Server.Documents.Commands.Replication;
+using Raven.Server.Documents.Replication;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Documents.Handlers.Processors.Replication
+{
+    internal abstract class AbstractReplicationHandlerProcessorForGetOutgoingFailureStats<TRequestHandler, TOperationContext> : AbstractHandlerProxyReadProcessor<ReplicationOutgoingsFailurePreview, TRequestHandler, TOperationContext>
+        where TOperationContext : JsonOperationContext
+        where TRequestHandler : AbstractDatabaseRequestHandler<TOperationContext>
+    {
+        protected AbstractReplicationHandlerProcessorForGetOutgoingFailureStats([NotNull] TRequestHandler requestHandler)
+            : base(requestHandler)
+        {
+        }
+
+        protected override RavenCommand<ReplicationOutgoingsFailurePreview> CreateCommandForNode(string nodeTag)
+        {
+            return new GetReplicationOutgoingsFailureInfoCommand(nodeTag);
+        }
+    }
+
+    public class ReplicationOutgoingsFailurePreview
+    {
+        public IDictionary<ReplicationNode, ConnectionShutdownInfo> OutgoingsFailureInfo;
+
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                ["Stats"] = new DynamicJsonArray(OutgoingsFailureInfo.Select(OutgoingFailureInfoToJson))
+            };
+        }
+
+        private DynamicJsonValue OutgoingFailureInfoToJson(KeyValuePair<ReplicationNode, ConnectionShutdownInfo> kvp)
+        {
+            return new DynamicJsonValue
+            {
+                ["Key"] = ReplicationNodeToJson(kvp.Key),
+                ["Value"] = kvp.Value
+            };
+        }
+    
+        private DynamicJsonValue ReplicationNodeToJson(ReplicationNode replicationNode)
+        {
+            var json = replicationNode.ToJson();
+            json[nameof(ReplicationNode)] = replicationNode.GetType().ToString();
```

Comment:
this looks weird.. I would expect this to be `Type` or something

Reply:
also, what does `GetType()` returns here? maybe we should use `FullName`?

Reply:
Yep.. looks weird to me too (that's from the original implementation). I'll happily change it to "_Type_" instead.
Also, seems like `GetType()` returns the Full name (for example, "_Raven.Client.Documents.Operations.Replication.InternalReplication_")

Reply:
Done